### PR TITLE
Refactor NSM

### PR DESCRIPTION
**Pull request by Bolodya1997**
## Description

Extract NSM healing logic to the separate class.
## Motivation and Context

**networkServiceManager** is a very complicated class. It holds a lot of logic so it is almost impossible to cover it with unit tests.

This refactoring has been done to make unit testing of connection healing feature possible.

## How Has This Been Tested?

- [x] Covered by existing integration testing
- [ ] Added integration testing to cover
- [x] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by haiodo** [comment]
@Bolodya1997 Please rebase on https://github.com/networkservicemesh/networkservicemesh/pull/989

File: controlplane/pkg/nsm/healer.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/model"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/serviceregistry"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+type networkServiceHealer interface {
+	healDstDown(healID string, connection *model.ClientConnection) bool
+	healDataplaneDown(healID string, connection *model.ClientConnection) bool
+	healDstUpdate(healID string, connection *model.ClientConnection) bool
```
Comment (lobkovilya) [inline controlplane/pkg/nsm/healer.go]: Based on discussion with @haiodo this method (and corresponding event also) could be renamed to `healRemoteDataplaneDown`

**Comment by nickolaev** [comment]
@Bolodya1997 shall we consider there are no functional changes but only code being moved around?

**Comment by Bolodya1997** [comment]
@nickolaev Yes

**Comment by nickolaev** [comment]
@haiodo please follow up with @Bolodya1997 for merging this pR>

File: controlplane/pkg/nsm/healer.go (hunk 1 above)
Comment (Bolodya1997) [reply controlplane/pkg/nsm/healer.go]: It seems to be reasonable. But I would rather prefer to postpone such "renaming refactoring" to the subsequent PR.

Comment (Bolodya1997) [reply controlplane/pkg/nsm/healer.go]: Done

Comment (haiodo) [reply controlplane/pkg/nsm/healer.go]: Let's do in this PR.

File: controlplane/pkg/apis/nsm/nsm.go [hunk 2]
```diff
@@ -51,11 +51,11 @@ type NetworkServiceClient interface {
...
 const (
-	HealState_DstDown       HealState = 1 // Destination is down, we need to restore it and re-program local Datplane.
-	HealState_SrcDown       HealState = 2 // Source is down, most probable will not happen yet.
-	HealState_DataplaneDown HealState = 3 // In case local Dataplane is down, we need to heal NSE/Remote NSM and Dataplane.
-	HealState_DstUpdate     HealState = 4 // Destination is updated, most probable because of Remote Dataplane is down, we need to re-program local dataplane.
-	HealState_DstNmgrDown   HealState = 5 // Destination is updated, most probable because of Remote Dataplane is down, we need to re-program local dataplane.
+	HealState_DstDown               HealState = 1 // Destination is down, we need to restore it and re-program local Datplane.
+	HealState_SrcDown               HealState = 2 // Source is down, most probable will not happen yet.
+	HealState_DataplaneDown         HealState = 3 // In case local Dataplane is down, we need to heal NSE/Remote NSM and Dataplane.
+	HealState_RemoteDataplaneUpdate HealState = 4 // Destination is updated, most probable because of Remote Dataplane is down, we need to re-program local dataplane.
```
Comment (haiodo) [inline controlplane/pkg/apis/nsm/nsm.go]: It should be HealState_RemoteDataplaneDown.

File: controlplane/pkg/nsm/healer.go [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+)
+
+type networkServiceHealer interface {
+	healDstDown(healID string, connection *model.ClientConnection) bool
+	healDataplaneDown(healID string, connection *model.ClientConnection) bool
+	healRemoteDataplaneDown(healID string, connection *model.ClientConnection) bool
+	healDstNmgrDown(healID string, connection *model.ClientConnection) bool
+}
+
+type healer struct {
```
Comment (haiodo) [inline controlplane/pkg/nsm/healer.go]: I would prefer nsmHealProcessor in naming and nsm_heal_processor.go

File: controlplane/pkg/apis/nsm/nsm.go (hunk 2 above)
Comment (Bolodya1997) [reply controlplane/pkg/apis/nsm/nsm.go]: Done : 5fdc2c1

File: controlplane/pkg/nsm/healer.go (hunk 3 above)
Comment (Bolodya1997) [reply controlplane/pkg/nsm/healer.go]: Done : 5fdc2c1

**Pull request merged**